org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -379,6 +407,26 @@ public Preferences setJavaHome(String javaHome) {
 		return this;
 	}
 
+	public Preferences setFormatterUrl(String formatterUrl) {
+		if (this.formatterUrl != null) {
+			File file = null;
+			try {
+				file = ResourceUtils.toFile(URIUtil.toURI(new URL(this.formatterUrl)));
+			} catch (MalformedURLException | URISyntaxException e) {
+				file = new File(this.formatterUrl);
+			} finally {
+				FormatterManager.unregisterWatcher(file);
```

Comment:
I really don't like seeing that call here. Should be done by the manager. It's also inconsistent with registering the watcher